lib/bundler/cli/outdated.rb

```diff
@@ -38,76 +45,129 @@ def run
       end
 
       Bundler.ui.info ""
+      outdated_gems_by_groups = {}
+      outdated_gems_list = []
 
-      out_count = 0
       # Loop through the current specs
       gemfile_specs, dependency_specs = current_specs.partition {|spec| current_dependencies.key? spec.name }
       [gemfile_specs.sort_by(&:name), dependency_specs.sort_by(&:name)].flatten.each do |current_spec|
         next if !gems.empty? && !gems.include?(current_spec.name)
 
         dependency = current_dependencies[current_spec.name]
+        next if dependency.nil?
 
-        if options["strict"]
-          active_spec = definition.specs.detect {|spec| spec.name == current_spec.name && spec.platform == current_spec.platform }
+        spec_groups = if options[:strict]
+          [definition.specs.detect {|spec| spec.name == current_spec.name && spec.platform == current_spec.platform }]
         else
-          active_specs = definition.index[current_spec.name].select {|spec| spec.platform == current_spec.platform }.sort_by(&:version)
-          if !current_spec.version.prerelease? && !options[:pre] && active_specs.size > 1
-            active_spec = active_specs.delete_if {|b| b.respond_to?(:version) && b.version.prerelease? }
-          end
-          active_spec = active_specs.last
-
-          if options[:major] || options[:minor] || options[:patch]
-            update_present = update_present_via_semver_portions(current_spec, active_spec, options)
-            active_spec = nil unless update_present
+          specs = definition.index[current_spec.name].select {|spec| spec.platform == current_spec.platform }.sort_by(&:version)
+          if !current_spec.version.prerelease? && !options[:pre] && specs.size > 1
+            specs.delete_if {|b| b.respond_to?(:version) && b.version.prerelease? }
+          else
+            specs
           end
         end
 
-        next if active_spec.nil?
-
+        active_spec = Bundler.definition.gem_version_promoter.sort_versions(dependency, spec_groups).last
```

Comment:
I don't understand this - this is an internal part of the resolution that's already happened inside Definition. It shouldn't be necessary to _re_-sort the versions ... we should just get the resolved dependencies out of the Definition like it used to do.